src/Validator/Validators/hdtXSDValidator.cpp

```
@@ -289,9 +291,38 @@ namespace hdt
 
 	// ---- public API ----
 
+	// Rewrites each schema violation's line from the EXPANDED document to the author's source line via the
+	// pattern source map, matching hdtSCHValidator. It does nothing when the file used no patterns (empty
+	// map); line-0 (parse/structural) violations are left untouched. The reader reports 1-based rows into
+	// the expanded text; pugixml pretty-prints one element per line, so the first '<' at or after a row's
+	// start is that element's tag, whose byte offset the map keys on.
+	static void remapXsdViolationsToSource(std::vector<XSDViolation>& violations, const PhysicsXmlSource& src)
+	{
+		if (src.sourceMap.ranges.empty())
```

Clang (14.0.6)

[warning] statement should be inside braces

(readability-braces-around-statements)